lib/pos_blockchain.go

```diff
@@ -60,66 +58,43 @@ func (bc *Blockchain) processBlockPoS(desoBlock *MsgDeSoBlock, currentView uint6
 	if err != nil {
 		return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem getting validator set: ")
 	}
-	// 1e. Validate QC
+	// 2. Validate QC
 	if err = bc.validateQC(desoBlock, validatorsByStake); err != nil {
```

Comment:
I don't have an easy solution for here. But just to call out that if the block belongs in the next epoch or if it's form an epoch far ahead in the future, we may not be able to validate its QC at all. The validator set for that epoch may be entirely different.

This is definitely a TODO for later. But I see a couple of options here:
1. We add utility to the UtxoView that can fetch the validator set given an arbitrary block height. If we can't fetch the validator set for the block, then we reject it (even if it later turns out to be a valid block).
2. We add the block to the block index *before* doing the QC validation such that even if we aren't able to fetch the validator set for the block, we can at least store it locally

Option 1 appears to be the simpler approach. LMK what you think.

Reply:
Option 1 is probably the better option. If it later turns out to be a valid block, we will call ProcessBlockPoS again on it to go through the whole process again *I think*